lib/reporters/json.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const fs = require('fs');
+const {resolve, isAbsolute, dirname} = require('path');
+
+function writeReport(fileName, data) {
+	try {
+		const dirPath = dirname(fileName);
+		if (!(fs.existsSync(dirPath) && fs.lstatSync(dirPath).isDirectory())) {
+			fs.mkdirSync(dirPath, {recursive: true});
+		}
+		fs.writeFileSync(fileName, data);
+		return true;
+	} catch (error) {
+		console.error(`Unable to write ${fileName}`);
+		console.error(error);
+		return false;
+	}
```

Comment:
Should the return values be removed since they're not being used?

```suggestion
	try {
		const dirPath = dirname(fileName);
		if (!(fs.existsSync(dirPath) && fs.lstatSync(dirPath).isDirectory())) {
			fs.mkdirSync(dirPath, {recursive: true});
		}
		fs.writeFileSync(fileName, data);
	} catch (error) {
		console.error(`Unable to write ${fileName}`);
		console.error(error);
	}
```